docs/user-guide/gpudeduplication.rst

```diff
@@ -5,164 +5,367 @@
-                   --seed 42 \
-                   --log-dir ./
-                   # --scheduler-file /path/to/file.json
+.. caution::
+    The CLI utilities are limited to Jsonl datasets and only work with GPU based backends.
+    For different dataset formats or backeds use the :ref:`exactdup_pyapi`.
 
+=========================================
+Fuzzy Deduplication
+=========================================
 
-  2. Buckets (Minhash Buckets)
-    - Input: Minhash directories
-    - Output: Buckets.parquet
-    - Example call:
+When removing near-duplicates within the corpus we perform fuzzy deduplication at the document level in order to remove documents that
+have high Jaccard similarity. Our approach closely resembles the approach described in `Smith et al., 2020 <https://arxiv.org/abs/2201.11990>`_.
 
-         .. code-block:: bash
+-----------------------------------------
+How It Works
+-----------------------------------------
 
-                 # same as `python minhash_lsh.py`
-                 minhash_buckets \
-                   --input-data-dirs /path/to/output_minhashes/dir1 /path/to/output_minhashes/dir2 \
-                   --output-bucket-dir /path/to/dedup_output \
-                   --input-minhash-field _minhash_signature \
-                   --input-json-id-field id_column_name \
-                   --minhash-length number_of_hashes \
-                   --num-bands num_bands \
-                   --buckets-per-shuffle 1 `#Value b/w [1-num_bands]. Higher is better but might lead to oom` \
-                   --log-dir ./
-                   # --scheduler-file /path/to/file.json
+This approach can essentially be split into the following stages:
+
+1. **Compute Minhashes**: The first stage involves computing `MinHash <https://en.wikipedia.org/wiki/MinHash>`_ Signatures on documents.
+   NeMo-Curator currently only supports character n-grams for MinHashing. An approximate metric of ~4.5 characters per word can be used to determine the n-gram size for users familiar with word based ngrams.
+2. **LSH** *(Locality Sensitive Hashing)*: Perform `LSH <https://en.wikipedia.org/wiki/Locality-sensitive_hashing>`_
+   to find candidate duplucates.
+
+3. **Buckets to Edgelist**: If not using the false positive check, we directly convert the LSH buckets to edges for the connected components computation.
+
+3. **False Positive Check** *(optional alternative to Buckets to Edgelist)*: Due to the approximate nature of the bucketing via MinHash + LSH
+   (`Leskovec et al., 2020 <http://infolab.stanford.edu/~ullman/mmds/ch3n.pdf>`_), NeMo-Curator provides the option to further
+   process each of the buckets by computing some pariwise jaccard similarity scores between documents in each bucket and filter out false positives that might have been hashed into the same bucket.
+
+  a. **Jaccard Map Buckets:** Since buckets generated by LSH can have high cardinality, we map multiple LSH buckets to a larger batches for
```

Comment:
```suggestion
  a. **Jaccard Map Buckets:** Since buckets generated by LSH can have high cardinality, we map multiple LSH buckets to larger batches for
```